lms/migrations/versions/e69454a18f38_add_file_table.py

```diff
@@ -0,0 +1,50 @@
+"""
+Add file table.
+
+Revision ID: e69454a18f38
+Revises: bb11b5b06274
+Create Date: 2021-06-15 12:08:04.564355
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "e69454a18f38"
+down_revision = "bb11b5b06274"
+
+
+def upgrade():
+    op.create_table(
+        "file",
+        sa.Column(
+            "created",
+            sa.DateTime(),
+            server_default=sa.func.now(),
+            nullable=False,
+        ),
+        sa.Column(
+            "updated",
+            sa.DateTime(),
+            server_default=sa.func.now(),
+            onupdate=sa.func.now(),
+            nullable=False,
+        ),
+        sa.Column("id", sa.Integer(), autoincrement=True, primary_key=True),
+        sa.Column(
+            "application_instance_id",
+            sa.Integer(),
+            sa.ForeignKey("application_instances.id", ondelete="cascade"),
+            nullable=False,
+        ),
+        sa.Column("type", sa.String(), nullable=False),
+        sa.Column("lms_id", sa.String(), nullable=False),
+        sa.Column("course_id", sa.String()),
+        sa.Column("name", sa.String()),
+        sa.Column("size", sa.Integer()),
+        sa.UniqueConstraint("application_instance_id", "lms_id", "type", "course_id"),
```

Comment:
I don't have strong opinions about this but if you have handwriting this migration you might have bypassed the naming conventions thingy:

https://github.com/hypothesis/lms/blob/2474c2f61ad58cbe2ead2f90fc11f8886c4eb0cd/lms/db/__init__.py#L76

that would explicitly add a `name=` to the FK and constraint.

Reply:
This seems to result in the same table, what's the difference?